Update TrueNAS Scale documentation

DissonanceTrip opened the pull request:
Updated truenas-scale.md to match a tested and verified method of deploying GameVault on TrueNAS Scale v24.10.

TrueNAS Scale v24.10 (Electric Eel) changes the app platform to Docker - this change effectively depreciates any TrueCharts Apps, and existing documentation will not work for current versions of TrueNAS Scale. 

Attempted to honor the formatting of the original documentation, but adjusted line breaks in a few places to improve readability.

Most images have been replaced, and are referencing my publicly available images. Please feel free to pull these, upload to your systems, and change the references as needed.

<3

Alfagun74 commented:
Hey! If it's not too much trouble, could you upload the images directly to [this folder](https://github.com/Phalcode/gamevault-docs/tree/master/static/img/docs/setup/scale)? I’ll be happy to merge after.

Also, do you think it makes sense to keep the old Truecharts documentation for users of the older system instead of replacing it?

DissonanceTrip commented:
Added referenced images to the specified folder and updated the name of the referenced images in the truenas-scale.md file.

As for keeping the old TrueCharts documentation - yeah I was thinking about that after submitting the PR. Might be worth having both for a while since I'm sure some people are going to hang onto older Scale versions to avoid rebuilding apps.

DissonanceTrip commented:
Ok - now that I'm not sleep deprived, I can update this PR to include both the original truenas_scale.md as well as my new doc for v24.10 and newer as a separate file. 

Would you prefer that my new version take the place of the main truenas_scale.md file and rename the old version something like truenas_scale_legacy.md? Or, would you prefer to combine them both into one doc or something else?

DissonanceTrip commented:
Changes:

- truenas-scale.md is my updated documentation for TrueNAS Scale version 24.10 or newer. 
- truenas-scale-legacy.md is the original documentation for TrueNAS Scale versions prior to 24.10 (changed name)
- Updated setup.md to include truenas-scale-legacy in position 7 (right above "Native" but below everything else)
- Updated both truenas-scale.md and truenas-scale-legacy.md to reference each other with note of which to use depending on your version of TrueNAS Scale


If you would prefer that there be ONE doc for TrueNAS Scale, and it contains both main and legacy information, let me know and I can make that change.

DissonanceTrip commented:
Updated PR name to better describe the proposed changes.

Updated truenas-scale.md (documentation for version 24.10 and newer) to match link method that the rest of the documentation uses. Updated image links to point to the correct location for the new images I uploaded.

Updated truenas-scale-legacy.md (documentation for versions prior to 24.10) to match link method that the rest of the documentation uses. 

This should actually be ready to merge now as far as I can tell! Let me know if I missed anything or if you would like any changes :)
